syntax = "proto3";

message Empty {
}

message TrainingConfig {
  string training_data_filename = 1; // .npz file
  string training_labels_filename = 2; // .npz file
  int32 epochs = 3;
  int32 batch_size = 4;
  double validation_ratio = 5;
  string model_filename = 6;
}

service NewsTrainer {
  rpc startTraining(Empty) returns(TrainingConfig);
}
